### PR TITLE
BOM-1597 : Add Python 3.8 Support

### PR DESCRIPTION
**Pull request by mzulqarnain1**
- Added Python 3.8 support
- Removed Python versions other than 3.5 and 3.8
- Upgraded dependecies
- Updated classifiers and bumped version

Relevant JIRA : https://openedx.atlassian.net/browse/BOM-1597

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,21 +3,17 @@ sudo: false
...
   - TOX_ENV=django22
-  - TOX_ENV=django21
-  - TOX_ENV=django20
-  - TOX_ENV=django111
 
 matrix:
   exclude:
     - python: 3.5
-      env: TOX_ENV=django3
+      env: TOX_ENV=django30
```
Comment (mraarif) [inline .travis.yml:16]: as per Jeremy, we shouldn't add Django 3.0 in Travis builds just yet

Comment (mzulqarnain1) [reply .travis.yml:16]: It was already added, I just changed it from 3 to 30. Should I remove it?

Comment (mraarif) [reply .travis.yml:16]: as described [here](https://openedx.atlassian.net/wiki/spaces/AC/pages/1368064245/Post-Upgrade+Test+Matrix+Cleanup), it shouldn't have been added in the first place

File: .travis.yml [hunk 2]
```diff
@@ -3,21 +3,17 @@ sudo: false
...
 
 before_install:
   - "pip install -U pip"
```
Comment (morenol) [inline .travis.yml:19]: I think that we should use here pip==20.0.2

**Pull request merged**